error_writer.go

```diff
@@ -94,7 +98,10 @@ func (w *ErrorWriter) IsSupported(request *http.Request) bool {
 //
 // Write does not read or close the request body.
 func (w *ErrorWriter) Write(response http.ResponseWriter, request *http.Request, err error) error {
-	ctype := request.Header.Get(headerContentType)
+	ctype, _, parseErr := mime.ParseMediaType(request.Header.Get(headerContentType))
+	if parseErr != nil {
+		return parseErr
+	}
```

Comment:
Two things:

1. Can we extract this repeated block into a utility function?
2. We should also be considering charset here - we shouldn't send UTF-8 if the client has requested `application/json; charset=shift-jis`.